### ISSUE TITLE
Suggests as dependency causes failure Columns must be 1d atomic vectors

### ISSUE THREAD

**Issue by dpastoor** [r-lib/pkgdepends#66]
```r
tmp <- pkgdepends::remotes$new("tidyverse", 
    lib = "lib", config = list(cache_dir = "pkgdir", 
dependencies = c("Depends", 
                 "Imports",
                 "LinkingTo",
                 "Suggests")))

tmp$solve()
# ✔ Resolved 1/1 direct refs and 1304/1304 dependencies   
tmp$get_install_plan() # fails
```

```
Error: Columns `package`, `type`, `version`, `direct`, `file` must be 1d atomic vectors or lists
```

I looked around and saw you recently added support for `dependencies = TRUE`, which is actually what I wanted, and that does work, as do the defaults.

```
Session info ------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.2 (2017-09-28)
 system   x86_64, darwin17.0.0        
 ui       RStudio (1.1.408)           
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       America/New_York            
 date     2018-01-05                  

Packages ----------------------------------------------------------------------------
 package        * version    date       source                              
 assertthat       0.2.0      2017-04-11 CRAN (R 3.4.2)                      
 async            0.0.0.9000 2018-01-05 Github (r-lib/async@93cfc23)        
 backports        1.1.2      2017-12-13 CRAN (R 3.4.2)                      
 base           * 3.4.2      2017-10-08 local                               
 BH               1.65.0-1   2017-08-24 CRAN (R 3.4.2)                      
 bindr            0.1        2016-11-13 CRAN (R 3.4.2)                      
 bindrcpp       * 0.2        2017-06-17 CRAN (R 3.4.2)                      
 bitops           1.0-6      2013-08-17 CRAN (R 3.4.2)                      
 broom            0.4.2      2017-02-13 CRAN (R 3.4.2)                      
 callr            1.0.0.9000 2018-01-05 Github (r-lib/callr@550fa6b)        
 cellranger       1.1.0      2016-07-27 CRAN (R 3.4.2)                      
 cli              1.0.0      2017-11-05 CRAN (R 3.4.2)                      
 colorspace       1.3-2      2016-12-14 CRAN (R 3.4.2)                      
 compiler         3.4.2      2017-10-08 local                               
 crayon           1.3.4      2017-09-16 CRAN (R 3.4.2)                      
 curl             3.1        2017-12-12 CRAN (R 3.4.2)                      
 datasets       * 3.4.2      2017-10-08 local                               
 debugme          1.1.0      2017-10-22 CRAN (R 3.4.2)                      
 desc             1.1.1.9002 2018-01-05 Github (r-lib/desc@ef0ce71)         
 devtools         1.13.3     2017-08-02 CRAN (R 3.4.2)                      
 digest           0.6.13     2017-12-14 CRAN (R 3.4.2)                      
 dplyr          * 0.7.4      2017-09-28 CRAN (R 3.4.2)                      
 evaluate         0.10.1     2017-06-24 CRAN (R 3.4.2)                      
 filelock         0.0.0.9000 2018-01-05 Github (r-lib/filelock@669108c)     
 forcats          0.2.0      2017-01-23 CRAN (R 3.4.2)                      
 foreign          0.8-69     2017-06-22 CRAN (R 3.4.2)                      
 ggplot2        * 2.2.1      2016-12-30 CRAN (R 3.4.2)                      
 glue             1.2.0      2017-10-29 cran (@1.2.0)                       
 graphics       * 3.4.2      2017-10-08 local                               
 grDevices      * 3.4.2      2017-10-08 local                               
 grid             3.4.2      2017-10-08 local                               
 gtable           0.2.0      2016-02-26 CRAN (R 3.4.2)                      
 haven            1.1.0      2017-07-09 CRAN (R 3.4.2)                      
 hms              0.4.0      2017-11-23 CRAN (R 3.4.2)                      
 htmltools        0.3.6      2017-04-28 CRAN (R 3.4.2)                      
 httpuv           1.3.5      2017-07-04 cran (@1.3.5)                       
 httr             1.3.1      2017-08-20 CRAN (R 3.4.2)                      
 jsonlite         1.5        2017-06-01 CRAN (R 3.4.2)                      
 knitr            1.18       2017-12-27 CRAN (R 3.4.2)                      
 later          * 0.6.0.9001 2018-01-01 Github (r-lib/later@75801a4)        
 lattice          0.20-35    2017-03-25 CRAN (R 3.4.2)                      
 lazyeval         0.2.1      2017-10-29 cran (@0.2.1)                       
 lpSolve          5.6.13     2015-09-19 CRAN (R 3.4.2)                      
 lubridate        1.6.0      2016-09-13 CRAN (R 3.4.2)                      
 magrittr         1.5        2014-11-22 CRAN (R 3.4.2)                      
 memoise          1.1.0      2017-10-20 Github (hadley/memoise@d63ae9c)     
 methods        * 3.4.2      2017-10-08 local                               
 microbenchmark   1.4-2.1    2015-11-25 CRAN (R 3.4.2)                      
 mime             0.5        2016-07-07 CRAN (R 3.4.2)                      
 mnormt           1.5-5      2016-10-15 CRAN (R 3.4.2)                      
 modelr           0.1.1      2017-07-24 CRAN (R 3.4.2)                      
 munsell          0.4.3      2016-02-13 CRAN (R 3.4.2)                      
 nlme             3.1-131    2017-02-06 CRAN (R 3.4.2)                      
 parallel         3.4.2      2017-10-08 local                               
 pillar           1.0.1      2017-11-27 CRAN (R 3.4.2)                      
 pkgbuild         0.0.0.9000 2018-01-05 Github (r-lib/pkgbuild@ce7f6d1)     
 pkgconfig        2.0.1      2017-03-21 CRAN (R 3.4.2)                      
 pkgdepends       0.0.0.9000 2018-01-05 Github (r-lib/pkgdepends@52c7261)   
 plyr             1.8.4      2016-06-08 CRAN (R 3.4.2)                      
 prettyunits      1.0.2      2015-07-13 CRAN (R 3.4.2)                      
 processx         2.0.0.1    2017-07-30 CRAN (R 3.4.2)                      
 progress         1.1.2.9000 2018-01-05 Github (r-lib/progress@3023c0b)     
 psych            1.7.8      2017-09-09 CRAN (R 3.4.2)                      
 purrr          * 0.2.4      2017-10-18 CRAN (R 3.4.2)                      
 R6               2.2.2      2017-06-17 CRAN (R 3.4.2)                      
 rappdirs         0.3.1      2016-03-28 CRAN (R 3.4.2)                      
 Rcpp             0.12.14    2017-11-23 CRAN (R 3.4.2)                      
 RCurl            1.95-4.8   2016-03-01 CRAN (R 3.4.2)                      
 readr          * 1.1.1      2017-05-16 CRAN (R 3.4.2)                      
 readxl           1.0.0      2017-04-18 CRAN (R 3.4.2)                      
 rematch2         2.0.1      2017-06-20 CRAN (R 3.4.2)                      
 remotes          1.1.0.9000 2017-11-14 Github (MangoTheCat/remotes@2363c85)
 reshape2         1.4.2      2016-10-22 CRAN (R 3.4.2)                      
 RJSONIO          1.3-0      2014-07-28 CRAN (R 3.4.2)                      
 rlang            0.1.6.9000 2018-01-01 Github (tidyverse/rlang@383a1e1)    
 rmarkdown        1.8        2017-11-17 CRAN (R 3.4.2)                      
 rprojroot        1.3-2      2018-01-03 CRAN (R 3.4.2)                      
 rsconnect        0.8.5      2017-08-23 CRAN (R 3.4.2)                      
 rstudioapi       0.7        2017-09-07 CRAN (R 3.4.2)                      
 rvest            0.3.2      2016-06-17 CRAN (R 3.4.2)                      
 scales           0.5.0      2017-08-24 CRAN (R 3.4.2)                      
 shiny          * 1.0.5      2017-08-23 cran (@1.0.5)                       
 stats          * 3.4.2      2017-10-08 local                               
 stringi        * 1.1.6      2017-11-17 CRAN (R 3.4.2)                      
 stringr          1.2.0      2017-02-18 CRAN (R 3.4.2)                      
 tibble         * 1.4.1      2017-12-25 CRAN (R 3.4.2)                      
 tidyr          * 0.7.2      2017-10-16 CRAN (R 3.4.2)                      
 tidyverse      * 1.1.1      2017-01-27 CRAN (R 3.4.2)                      
 tools            3.4.2      2017-10-08 local                               
 utils          * 3.4.2      2017-10-08 local                               
 uuid             0.1-2      2015-07-28 CRAN (R 3.4.2)                      
 withr            2.1.1      2017-12-19 CRAN (R 3.4.2)                      
 xml2             1.1.1      2017-01-24 CRAN (R 3.4.2)                      
 xtable           1.8-2      2016-02-05 cran (@1.8-2)                       
 yaml             2.1.16     2017-12-12 CRAN (R 3.4.2)     
```

**Comment by gaborcsardi** [comment]
Thanks, will take a look soon!

**Comment by dpastoor** [comment]
great - I will be digging around these packages a lot in the coming couple weeks so hopefully will be able to contribute some bug reports/tests/docs upstream.

**Comment by gaborcsardi** [comment]
This happens because some of the suggested packages are not available on CRAN and BioC (any more):

```r
tmp <- pkgdepends::remotes$new("tidyverse",
     lib = "lib", config = list(cache_dir = "pkgdir",
     dependencies = c("Depends", "Imports", "LinkingTo", "Suggests")))
tmp$resolve()
tmp$get_resolution()
```

```
#>  🄿   RESOLUTION, 1 refs, resolved in 8.9s 
#> ─────────────────────────────────────
#>   tidyverse
#> Dependencies:
#>   standard:
#>     abind, acepack, ACME, actuar, ActuDistns, ada, ade4, ade4TkGUI,
#>     adegenet, adegraphics, adehabitatLT, adehabitatMA, adephylo,
#> 
#> ...
#> 
#>     yeastNagalakshmi, Zelig, ZeligChoice, ZeligEI, zeligverse, zipcode,
#>     zipfR, zlibbioc, zoo, ztable
#> Errors:
#>   ActuDistns: Can't find CRAN/BioC package ActuDistns
#>   cacheSweave: Can't find CRAN/BioC package cacheSweave
#>   glmmADMB: Can't find CRAN/BioC package glmmADMB
#>   Rcampdf: Can't find CRAN/BioC package Rcampdf
#>   rcom: Can't find CRAN/BioC package rcom
#>   Rcompression: Can't find CRAN/BioC package Rcompression
#>   RcppOctave: Can't find CRAN/BioC package RcppOctave
#>   RDCOMClient: Can't find CRAN/BioC package RDCOMClient
#>   rrdf: Can't find CRAN/BioC package rrdf
#>   spDataLarge: Can't find CRAN/BioC package spDataLarge
#>   statisticalModeling: Can't find CRAN/BioC package statisticalModeling
#>   Sxslt: Can't find CRAN/BioC package Sxslt
#>   tm.lexicon.GeneralInquirer: Can't find CRAN/BioC package tm.lexicon.GeneralInquirer
```
`solve()` should give a better error, of course. An we would also need a way to ignore resolution errors and still proceed with `solve()`.

**Comment by gaborcsardi** [comment]
Opened separate issues for these: #69, #70, so I am closing this one.
